legend-engine-xts-data-space/legend-engine-xt-data-space-grammar/src/main/antlr4/org/finos/legend/engine/language/pure/grammar/from/antlr4/DataSpaceParserGrammar.g4

```diff
@@ -155,6 +156,8 @@ executablePath:                     DATA_SPACE_EXECUTABLE COLON qualifiedName SE
 ;
 executableTemplateQuery:            DATA_SPACE__TEMPLATE_QUERY COLON combinedExpression SEMI_COLON
 ;
+executableTemplateQueryId:          DATA_SPACE__TEMPLATE_QUERY__ID COLON (VALID_STRING | INTEGER | DECIMAL) SEMI_COLON
```

Comment:
Just a heads-up
123 will be prioritized as INTEGER 
2d will be treated as DECIMAL 
even if we define fragment ValidString  as  (Letter | Digit | '_' ) (Letter | Digit | '_' | '$')*
to allow pure digits and or some digit + 'd' 
we have to specify INTEGER | DECIMAL

Reply:
Besides, ID is designed to not have quotes.  Pure digits are allowed e.g. id: 123
only string follows this pattern, (Letter | Digit | '' ) (Letter | Digit | '' | '$')*, will be allowed, which means id shouldn't contain white space 


a heads-up, as I pointed it out above, digits + 'd'/'D' will be prioritized as decimal.